### PR TITLE
EVM

### PR DESCRIPTION
**Pull request by andriitserkovnyi**
## PR Description

Docs for the EVM feature

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

**Review by hatemzidi-pro — COMMENTED** [review]
overall good for me.

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    * `scheduler`: used by the scheduler.
+
+* `grant`: Defines access permissions to variables. Possible values:
+    * `public`: readable and writable
+    * `limited`: readable
+    * `internal`: hidden
+
+Path examples:
+
+* /fashion_club_store/staging/config/common/limited/composer_pass
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:36]:
```suggestion
* `/fashion_club_store/staging/config/common/limited/composer_pass`
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+* /deans_jeans/prod/config/app/public/mail_host
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:38]:
```suggestion
* `/deans_jeans/prod/config/app/public/mail_host`
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+## Variable path hierarchy
+
+Path hierarchy is needed to cover the cases when several variables with the same name are declared. If several variables with the same name are declared, the variable with a higher priority applies. The following rules define the priority of variables:
+
+1. For any `type` and `bucket`, the priority is `public` > `limited` > `internal`.
+2. Foy any `bucket`, the priority is `bucket` > `common`.
+3. For any variable with the same name, the priority is `secret` > `config`.
+
+The following variables are arranged from lower to higher priority:
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:49]: Shouldn't lines 51-73 be formatted as code font?

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+* /{project}/{environment}/secret/{app | scheduler}/public/{variable_name}
+
+## Add variables
+
+The following sections describe how to add parameters and secrets for different resources.
+
+{% info_block warningBox "Propagation of variables" %}
+
+To make variables available in your Jenkins instance, we need to terraform your added or changed variables. To do this, create a [support case](https://docs.spryker.com/docs/scos/user/intro-to-spryker/support/how-to-use-the-support-portal.html#plattform-change-requests).
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:81]:
```suggestion
To make variables available in your Jenkins instance, we need to terraform your added or changed variables. To do this, create a [support case](/docs/scos/user/intro-to-spryker/support/how-to-use-the-support-portal.html#plattform-change-requests).
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+{% endinfo_block %}
+
+### Add parameters to all resource types
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:87]:
```suggestion
1. In the AWS Management Console, go to **Services&nbsp;<span aria-label="and then">></span> Parameter Store**.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+2. In the **My parameters** pane, click **Create parameter**.
+    This opens the **Create parameter** page.
+3. For **Name**, enter `/{project}/{environment}/config/common/public/{variable_name}`.
+    Make sure to replace the placeholders based on your requirements.
+4. Optional: Enter a **Description**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:92]: According to google style guide, this should be rephrased to something like (see https://developers.google.com/style/ui-elements#refer-to-ui-elements):
```suggestion
4. Optional: For **Description**, enter the description of the variable.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+    This may be a note about what this variable is used for.
+5. Select a **Type** based on your requirements.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:94]: same: requires rephrasing

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
+2. In the **My parameters** pane, click **Create parameter**.
+    This opens the **Create parameter** page.
+3. For **Name**, enter `/{project}/{environment}/config/common/public/{variable_name}`.
+    Make sure to replace the placeholders based on your requirements.
+4. Optional: Enter a **Description**.
+    This may be a note about what this variable is used for.
+5. Select a **Type** based on your requirements.    
+6. Enter a **Value**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:95]: same: consider rephrasing

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+7. Click **Create parameter**.
+    This opens the **Parameter Store** page with a success message displayed.    
+8. Go to **Services** > **CodePipeline**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:98]:
```suggestion
8. Go to **Services**&nbsp;<span aria-label="and then">></span> **CodePipeline**.
```

Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:98]:
```suggestion
8. Go to **Services**&nbsp;<span aria-label="and then">></span> **CodePipeline**.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+9. On the **Pipelines** page, select the **NORMAL_Deploy_Spryker_{project}-{environemt}** pipeline.
+10. On the pipeline's page, click **Release change**.
+11. In the **Release change** window, click **Release**.
+    After the pipeline finishes running, the variable gets available for your application.
+
+### Add parameters to ECS applications
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:125]:
```suggestion
1. In the AWS Management Console, go to **Services&nbsp;<span aria-label="and then">></span> Parameter Store**.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+    This opens the **Parameter Store** page with a success message displayed.    
+8. Go to **Services** > **CodePipeline**.
+9. On the **Pipelines** page, select the **ECS-updater-{project}-{environemt}** pipeline.
+10. On the pipeline's page, click **Release change**.
+11. In the **Release change** window, click **Release**.
+    After the pipeline finishes running, the variable gets available for your application.
+
+### Adding secrets to ECS applications
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:144]:
```suggestion
1. In the AWS Management Console, go to **Services&nbsp;<span aria-label="and then">></span> Parameter Store**.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+    This opens the **Create parameter** page.
+3. For **Name**, enter `/{project}/{environment}/secret/app/public/{variable_name}`.
+    Make sure to replace the placeholders based on your requirements.
+4. Optional: Enter a **Description**.
+    This may be a note about what this variable is used for.
+5. For **Type**, select **SecureString**.
+6. Enter a **Value**.
+7. Click **Create parameter**.
+    This opens the **Parameter Store** page with a success message displayed.    
+8. Go to **Services** > **CodePipeline**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:155]:
```suggestion
8. Go to **Services**&nbsp;<span aria-label="and then">></span> **CodePipeline**.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+9. On the **Pipelines** page, select the **ECS-updater-{project}-{environemt}** pipeline.
+10. On the pipeline's page, click **Release change**.
+11. In the **Release change** window, click **Release**.
+    After the pipeline finishes running, the variable gets available for your application.    
+
+### Add parameters and secrets to Scheduler
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:163]:
```suggestion
1. In the AWS Management Console, go to **Services&nbsp;<span aria-label="and then">></span> Parameter Store**.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+    * Variable: `/{project}/{environment}/config/scheduler/public/{variable_name}`
+    * Secret: `/{project}/{environment}/secret/scheduler/public/{variable_name}`
+        Make sure to replace the placeholders based on your requirements.
+4. Optional: Enter a **Description**.
+    This may be a note about what this variable is used for.
+5. For **Type**, select **SecureString**.
+6. Enter a **Value**.
+7. Click **Create parameter**.
+    This opens the **Parameter Store** page with a success message displayed.    
+8. Go to **Services** > **CodePipeline**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:176]:
```suggestion
8. Go to **Services&nbsp;<span aria-label="and then">></span> CodePipeline**.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+* /fashion_club_store/staging/config/common/limited/composer_pass
+
+* /deans_jeans/prod/config/app/public/mail_host
+
+
+## Variable path hierarchy
+
+Path hierarchy is needed to cover the cases when several variables with the same name are declared. If several variables with the same name are declared, the variable with a higher priority applies. The following rules define the priority of variables:
+
+1. For any `type` and `bucket`, the priority is `public` > `limited` > `internal`.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:45]: consider adding `&nbsp;<span aria-label="and then">></span>`

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+2. Foy any `bucket`, the priority is `bucket` > `common`.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:46]: consider adding `&nbsp;<span aria-label="and then">></span>`

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+    This opens the **Parameter Store** page with a success message displayed.    
+8. Go to **Services** > **CodePipeline**.
+9. On the **Pipelines** page, select the **NORMAL_Deploy_Spryker_{project}-{environemt}** pipeline.
+10. On the pipeline's page, click **Release change**.
+11. In the **Release change** window, click **Release**.
+    After the pipeline finishes running, the variable gets available for your application.
+
+### Add secrets to all resource types
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:106]:
```suggestion
1. In the AWS Management Console, go to **Services&nbsp;<span aria-label="and then">></span> Parameter Store**.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 18]
```diff
@@ -0,0 +1,180 @@
...
+2. In the **My parameters** pane, click **Create parameter**.
+    This opens the **Create parameter** page.
+3. For **Name**, enter `/{project}/{environment}/secret/common/public/{variable_name}`.
+    Make sure to replace the placeholders based on your requirements.
+4. Optional: Enter a **Description**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:111]: same: consider rephrasing

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 19]
```diff
@@ -0,0 +1,180 @@
...
+    This may be a note about what this variable is used for.
+5. For **Type**, select **SecureString**.
+6. Enter a **Value**.
+7. Click **Create parameter**.
+    This opens the **Parameter Store** page with a success message displayed.    
+8. Go to **Services** > **CodePipeline**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:117]:
```suggestion
8. Go to **Services&nbsp;<span aria-label="and then">></span> CodePipeline**.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 20]
```diff
@@ -0,0 +1,180 @@
...
+    After the pipeline finishes running, the variable gets available for your application.
+
+### Add parameters to ECS applications
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
+2. In the **My parameters** pane, click **Create parameter**.
+    This opens the **Create parameter** page.
+3. For **Name**, enter `/{project}/{environment}/config/app/public/{variable_name}`.
+    Make sure to replace the placeholders based on your requirements.
+4. Optional: Enter a **Description**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:130]: same: consider rephrasing

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 21]
```diff
@@ -0,0 +1,180 @@
...
+    This may be a note about what this variable is used for.
+5. Select a **Type** based on your requirements.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:132]: same: consider rephrasing

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 22]
```diff
@@ -0,0 +1,180 @@
...
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
+2. In the **My parameters** pane, click **Create parameter**.
+    This opens the **Create parameter** page.
+3. For **Name**, enter `/{project}/{environment}/config/app/public/{variable_name}`.
+    Make sure to replace the placeholders based on your requirements.
+4. Optional: Enter a **Description**.
+    This may be a note about what this variable is used for.
+5. Select a **Type** based on your requirements.    
+6. Enter a **Value**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:133]: same: consider rephrasing

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 23]
```diff
@@ -0,0 +1,180 @@
...
+7. Click **Create parameter**.
+    This opens the **Parameter Store** page with a success message displayed.    
+8. Go to **Services** > **CodePipeline**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:136]:
```suggestion
8. Go to **Services&nbsp;<span aria-label="and then">></span> CodePipeline**.
```

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 24]
```diff
@@ -0,0 +1,180 @@
...
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
+2. In the **My parameters** pane, click **Create parameter**.
+    This opens the **Create parameter** page.
+3. For **Name**, enter `/{project}/{environment}/secret/app/public/{variable_name}`.
+    Make sure to replace the placeholders based on your requirements.
+4. Optional: Enter a **Description**.
+    This may be a note about what this variable is used for.
+5. For **Type**, select **SecureString**.
+6. Enter a **Value**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:152]: same: consider rephrasing

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 25]
```diff
@@ -0,0 +1,180 @@
...
+### Add parameters and secrets to Scheduler
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
+2. In the **My parameters** pane, click **Create parameter**.
+    This opens the **Create parameter** page.
+3. For **Name**, enter one of the following:
+    * Variable: `/{project}/{environment}/config/scheduler/public/{variable_name}`
+    * Secret: `/{project}/{environment}/secret/scheduler/public/{variable_name}`
+        Make sure to replace the placeholders based on your requirements.
+4. Optional: Enter a **Description**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:170]: same: consider rephrasing

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 26]
```diff
@@ -0,0 +1,180 @@
...
+    This may be a note about what this variable is used for.
+5. For **Type**, select **SecureString**.
+6. Enter a **Value**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:173]: same: consider rephrasing

File: docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md [hunk 27]
```diff
@@ -0,0 +1,180 @@
...
+
+1. In the AWS Management Console, go to **Services > Parameter Store**.
+2. In the **My parameters** pane, click **Create parameter**.
+    This opens the **Create parameter** page.
+3. For **Name**, enter `/{project}/{environment}/secret/common/public/{variable_name}`.
+    Make sure to replace the placeholders based on your requirements.
+4. Optional: Enter a **Description**.
+    This may be a note about what this variable is used for.
+5. For **Type**, select **SecureString**.
+6. Enter a **Value**.
```
Comment (VadymSachenko) [inline docs/cloud/dev/spryker-cloud-commerce-os/add-variables.md:114]: same: consider rephrasing

**Review by VadymSachenko — CHANGES_REQUESTED** [review]

**Review by VadymSachenko — DISMISSED** [review]

**Review by VadymSachenko — APPROVED** [review]

**Pull request merged**